icefoganalytics/travel-authorization
# Build UI Travel Desk Request "submitted" State in Travel -> Request Tab

Fixes https://github.com/icefoganalytics/travel-authorization/issues/199

Relates to:
- https://github.com/icefoganalytics/travel-authorization/issues/167

# Context

**Is your feature request related to a problem? Please describe.**
Add a read-only version of travel desk travel request and dependent models - (travel desk travel request -> "submitted" state)

**Describe the solution you'd like**
Add read-only versions of all components on Travel Authorization -> Request tab.
Trigger read-only mode when Travel Desk Request is in a "submitted" state.

**Describe alternatives you've considered**
Use existing components; rejected due to difficulty of using them.

**Additional context**
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/158fbe68-c3e1-4e36-acf5-41a6c72d786a)

# Implementation

Add read-only views for all components from the edit page.
> The overall styling of this page is pretty ugly, but I haven't had a chance to clean it up yet. Probably needs a complete re-style.
Auto-compute the title width for the special inline title card component.

# Screenshots

http://localhost:8080/my-travel-requests/137/request
![image](https://github.com/user-attachments/assets/612aba60-23d8-4b2c-9329-9967cb4c5489)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to the My Travel Requests page via the top right drop down nav.
5. Create a new travel Authorization
6. Fill it in, and generate an estimate.
7. Submit it to yourself as the supervisor.
8. Go to the Manager View via the top drop down nav.
9. Find you request in the "Pending Approvals" card and click on it.
10. Scroll to the bottom and approve it.
11. Go back to the "My Travel Requests" page and click on the Submit Travel Desk Request call to action button on the new travel authorization you have created.
12. Add some of everything and then submit.
13. Check that the page now has read-only views of all the data.
14. Also check that the inline titles have the correct margin around the outline; its now auto-generated.
